datafusion/physical-expr/src/physical_expr.rs

```diff
@@ -51,23 +49,21 @@ pub trait PhysicalExpr: Send + Sync + Display + Debug {
         batch: &RecordBatch,
         selection: &BooleanArray,
     ) -> Result<ColumnarValue> {
-        if selection.iter().all(|b| b == Some(true)) {
+        let filter = match selection.null_count() {
+            0 => BooleanArray::from(selection.data().clone()),
+            _ => prep_null_mask_filter(selection),
+        };
+
+        let filter_count = filter
+            .values()
+            .count_set_bits_offset(selection.offset(), selection.len());
+
+        if filter_count == selection.len() {
             return self.evaluate(batch);
```

Comment:
This logic is already handled upstream, I don't see a reason to duplicate it here?

Reply:
See comment. The fasth path avoids calling `scatter` when the `selection` array is all-true. I reused some code from upstream, as it is not public yet.
In theory, we could re-use the filter count calculation, but we can't currently wire them together.

I am also OK with removing this path, for making the code a bit simpler.